packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1465,6 +1515,11 @@ export class KeyringController extends BaseController<
             options.createWithData,
           )) as SelectedKeyring;
         }
+      } else if ('id' in selector) {
+        const index = this.state.keyringsMetadata.findIndex(
+          (metadata) => metadata.id === selector.id,
+        );
+        keyring = this.#keyrings[index] as SelectedKeyring;
```

Comment:
Very very cosmetic: Not sure if that's better, but just wanted to share just in case:

IMO, we could have split `#getKeyringByIdOrDefault` even more like:

```ts
  #getKeyringById(keyringId: string): EthKeyring<Json> | undefined {
    const index = this.state.keyringsMetadata.findIndex(
      (metadata) => metadata.id === keyringId,
    );
    return this.#keyrings[index];
  }

  #getKeyringByIdOrDefault(keyringId?: string): EthKeyring<Json> | undefined {
    if (!keyringId) {
      return this.#keyrings[0] as EthKeyring<Json>;
    }

    return this.#getKeyringById(keyringId);
  }

```

```suggestion
        keyring = this.#getKeyringById(selector.id) as SelectedKeyring;
```

Reply:
Makes sense!